assets/src/edit-story/components/inspector/document/publish/index.js

```diff
@@ -99,6 +119,34 @@ function PublishPanel() {
     [updateStory]
   );
 
+  const generateCanvas = useCallback(async () => {
+    try {
+      const html2canvas = (await import('html2canvas')).default;
+      const canvas = await html2canvas(fullbleedContainer);
+      const blob = await new Promise((resolve, reject) =>
+        canvas.toBlob(
+          (value) => (value ? resolve(value) : reject(new Error())),
+          'image/jpeg'
+        )
+      );
+      const { id, media_details, source_url } = await uploadFile(
+        new File([blob], `web-story-${storyId}-cover-generated.jpg`, {
+          type: 'image/jpeg',
+        })
+      );
+      updateStory({
```

Comment:
After a video poster some meta data is generated to hide the image from the media library and automatically delete  the poster. I think we should do this here as well. 

See this snippet that adds the required meta data. 
https://github.com/google/web-stories-wp/blob/bd13cd643405a48bbcebacb9f519314cb47befed/assets/src/edit-story/app/media/utils/useUploadVideoFrame.js#L54-L66

The media is hidden here https://github.com/google/web-stories-wp/blob/9b3a1b2ebdfb8d0ba6f9eb006bdabfb1e05c2ba1/includes/Media.php#L223-L246

and videos are deleted here

https://github.com/google/web-stories-wp/blob/9b3a1b2ebdfb8d0ba6f9eb006bdabfb1e05c2ba1/includes/Media.php#L402-L425

To make this possible, some small tweaks to the PHP will need to be made and tests written. I am happy to add this PR if helpful. 

Thoughts @swissspidy

Reply:
My impression was that the generated cover image here is already hidden in the media library. Is that not the case?

Reply:
That is not hidden in the media library. See this. 

<img width="456" alt="Screenshot 2020-09-07 at 15 54 28" src="https://user-images.githubusercontent.com/237508/92399732-aead5a80-f122-11ea-84b4-c616c09e076f.png">

It would only hidden if we added that post meta.

Reply:
Good call. Doing the `updateMedia` call after upload prevents it from showing up in the gallery.

Reply:
@spacedmonkey So what we need is a way to delete orphaned cover images? Can you create a new issue for this to outline your thoughts?